src/go/rpk/pkg/cli/cmd/container/start.go

```diff
@@ -103,25 +117,28 @@ func startCluster(
 		return nil
 	}
 
-	log.Info("Downloading latest version of Redpanda")
-	err = common.PullImage(c)
-	if err != nil {
-		log.Debugf("Error trying to pull latest image: %v", err)
-
-		msg := "Couldn't pull image and a local one wasn't found either."
-		if c.IsErrConnectionFailed(err) {
-			msg += "\nPlease check your internet connection" +
-				" and try again."
-		}
-
-		present, checkErr := common.CheckIfImgPresent(c)
-
-		if checkErr != nil {
-			log.Debugf("Error trying to list local images: %v", err)
-
-		}
-		if !present {
-			return errors.New(msg)
+	log.Debug("Checking for a local image.")
+	present, checkErr := common.CheckIfImgPresent(c, image)
+	if checkErr != nil {
+		log.Debugf("Error trying to list local images: %v", err)
+	}
+	if !present {
```

Comment:
Keep in mind that `rpk container` is a tool for people who might not be very experienced with docker. So introducing flags to achieve 1:1 parity with the docker CLI isn't a goal.

The use case for the `--image` flag is testing, and shouldn't be used in any other situations, because it can lead to confusion and weird scenarios, like the errors that happened when `rpk container` used to pull `latest` instead of a matching version. The cool thing about `rpk container start` as it is is there's only one flag, `--nodes`. That's all you need to get a working cluster.